File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Controllers/TableViewControllers/TableViewCells/BtNewBugDetailsSelectTableViewCell.cs
using System;

using Foundation;
using UIKit;

namespace bugTrapKit
{
	public partial class BtNewBugDetailsSelectTableViewCell : BtNewBugDetailsTableViewCell
	{
		NSIndexPath loadingKey;

		UILabel initialsLabel;

		public BtNewBugDetailsSelectTableViewCell (IntPtr handle) : base(handle)
		{
			
		}

		public override void AwakeFromNib ()
		{
			base.AwakeFromNib();

			ImageView.Layer.CornerRadius = 22;
			ImageView.Layer.MasksToBounds = true;

			initialsLabel = new UILabel ();

			initialsLabel.TextColor = Colors.Theme;
			initialsLabel.TextAlignment = UITextAlignment.Center;
			initialsLabel.Font = Fonts.HelveticaNeue_Medium_22;
			initialsLabel.BackgroundColor = Colors.Clear;

			AddSubview(initialsLabel);

			ImageView.ContentMode = UIViewContentMode.ScaleAspectFit;
		}


		public override void LayoutSubviews ()
		{
			base.LayoutSubviews();

			initialsLabel.Frame = ImageView.Frame;
		}


		public override void SetData (CellData cellData)
		{
			var title = cellData.Get(DataKeys.Title);
			var detail = cellData.Get(DataKeys.Detail);
			var auxiliary = cellData.Get(DataKeys.Auxiliary);

			if (TextLabel != null) TextLabel.Text = title;
			if (DetailTextLabel != null) DetailTextLabel.Text = detail;
			if (initialsLabel != null) initialsLabel.Text = auxiliary;
		}


		public void SetData (NSIndexPath indexPath, CellData cellData, UIImage image = null)
		{
			loadingKey = indexPath;

			ImageView.Image = image;

			SetData(cellData);
		}


		public bool SetImage (UIImage image, NSIndexPath key)
		{
			var setImage = key == loadingKey;

			initialsLabel.Hidden = setImage;

			if (setImage) ImageView.Image = image;

			return setImage;
		}
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Controllers/TableViewControllers/TableViewCells/BtNewBugDetailsSelectTableViewCell.swift
//
//  BtNewBugDetailsSelectTableViewCell.swift
//  bugTrap
//
//  Created by Colby L Williams on 6/30/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import UIKit

class BtNewBugDetailsSelectTableViewCell: BtNewBugDetailsTableViewCell {

	var loadingKey: NSIndexPath?
	
	var initialsLabel = UILabel()
	
	override func awakeFromNib() {
		super.awakeFromNib()
		
		imageView?.layer.cornerRadius = 22.0
		imageView?.layer.masksToBounds = true
		
		initialsLabel.textColor = Colors.Theme.uiColor
		initialsLabel.textAlignment = .Center
		initialsLabel.font = UIFont(name: "HelveticaNeue-Medium", size: 20.0)
		initialsLabel.backgroundColor = Colors.Clear.uiColor
		
		addSubview(initialsLabel)
		
		imageView?.contentMode = UIViewContentMode.ScaleAspectFit
	}
	
	override func layoutSubviews() {
		super.layoutSubviews()
		
		if let imageViewFrame = imageView?.frame {
			initialsLabel.frame = imageViewFrame
		}
	}
	
	
	override func setData(title: String? = nil, detail: String? = nil, placeholder: String? = nil, auxiliary: String? = nil) {
		textLabel?.text = title ?? ""
		detailTextLabel!.text = detail ?? ""
		initialsLabel.text = auxiliary ?? ""
	}
	
	
	func setDataWithKey (indexPath: NSIndexPath, title: String? = nil, image: UIImage? = nil, detail: String? = nil, auxiliary: String? = nil) {
		
		loadingKey = indexPath
		
		imageView?.image = image
		
		setData(title, detail: detail, placeholder: nil, auxiliary: auxiliary)
	}


	func setImage(image: UIImage, key: NSIndexPath) -> Bool {
		
		initialsLabel.hidden = key == loadingKey
		
		if key == loadingKey {

//			if let f = imageView?.frame {
//				imageView?.frame = CGRect(x: f.origin.x - 12, y: f.origin.y - 12, width: f.width - 24, height: f.height - 24)
//			}
			
			imageView?.image = image
			
			return true
		}
		
		return false
    }
}